pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -126,6 +131,9 @@ public CompletableFuture<Boolean> asyncReload(String key, Boolean oldValue,
                         }
                     }
                 });
+
+        this.metadataStoreName = poolName + "-" + ID.getAndIncrement();
```

Comment:
I don't get this at all.
1. Do we have more than one metadata store in Pulsar Broker?
2. Why a name is derived from a thread pool name?
3. Why the name is actually an auto-generated name?

Reply:
1. there are more than 1 Metadata store, one for broker metadata, one for dynamic config.
you can see metadataStoreUrl and configurationMetadataStoreUrl in https://github.com/apache/pulsar/blob/master/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java
2. the metadata store name generated by the same rule with ThreadPool, keep align with ThreadPool
3. keep align with ThreadPool name

Reply:
I guess my question is this: If we have 2 types of metadata stores, one instance of each, why not have the metadata store name be just the type: "metadata-store" and "configuration-metadata-store" - why keep regenerating the name using an auto incremented ID?